Tests/Marketplace/search_and_install_packs.py

```diff
@@ -372,7 +374,18 @@ def search_pack_and_its_dependencies(client: demisto_client,
 
         current_packs_to_install = [pack_data]
         if dependencies:
-            current_packs_to_install.extend(dependencies)
+            # Check that the dependencies don't include a deprecated pack:
+            for dependency in dependencies:
+                pack_path = os.path.join(PACKS_DIR, dependency.get('id'))
+                if is_pack_deprecated(pack_path):
+                    logging.exception(f'Pack {pack_id} depends on pack {dependency.get("id")} which is a deprecated '
+                                      f'pack.')
+                    lock.acquire()
```

Comment:
If `SUCCESS_FLAG` is initialized as `True` and the threads can only set it to `False` there is no need to lock it since even in a race condition where two threads will try to set it to False at the same time - the result will be False in the end.